code/game/objects/structures/xeno.dm

```diff
@@ -733,16 +733,29 @@ TUNNEL
 	update_icon()
 
 /obj/effect/alien/resin/acidwell/Destroy()
+	creator = null
+	return ..()
+
+
+/obj/effect/alien/resin/acidwell/obj_destruction(damage_flag)
 	if(!QDELETED(creator) && creator.stat == CONSCIOUS && creator.z == z)
 		var/area/A = get_area(src)
 		if(A)
 			to_chat(creator, "<span class='xenoannounce'>You sense your acid well at [A.name] has been destroyed!</span>")
-	var/datum/effect_system/smoke_spread/xeno/acid/A = new(get_turf(src))
+
+	var/turf/open/T = get_turf(src)
+	if(!istype(T)) //We don't spawn gas on a closed turf.
+		return
+
+	if(!T.allow_construction) //We don't spawn gas on the dropship and other no-build tiles we shouldn't be on.
+		return
+
+	var/datum/effect_system/smoke_spread/xeno/acid/A = new(T)
```

Comment:
```suggestion
	var/datum/effect_system/smoke_spread/xeno/acid/A = new(get_turf(src))
```

Reply:
What? Isn't this going to spawn the acid gas as normal?